### PR TITLE
Avatar-Badge

### PR DESCRIPTION
**Pull request by ark2002**
- added avatar and badges components
- added main.css
- linked the avatar and badge pages from component page
- added a media query for mobile view in component page

### CODE REVIEW

File: components/avatar/avatar.css [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  padding: 1rem;
+}
+.avatar {
+  width: fit-content;
+  height: fit-content;
+  position: relative;
+}
+.avatar__img {
+  width: 57px;
+  height: 57px;
```
Comment (Mohasin-Haque) [inline components/avatar/avatar.css]: Please use relative units as much as you can.

File: components/avatar/avatar.html [hunk 2]
```diff
@@ -0,0 +1,117 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
```
Comment (Mohasin-Haque) [inline components/avatar/avatar.html]: You can add more meta descriptions to personalise it more. Like
<meta name="author" content="Your name" />
    <meta name="description" content="About your ui" />

File: components/badge/badge.html [hunk 3]
```diff
@@ -0,0 +1,60 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
```
Comment (Mohasin-Haque) [inline components/badge/badge.html]: same can be done here also

File: components/badge/badge.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            alt="AtroUI logo"
+            class="logo__img"
+          />
+          <h2 class="header__logo">Astro UI</h2>
+        </div>
+      </a>
+      <nav class="header_nav">
+        <ul>
+          <li><a href="#">Documentation</a></li>
+          <li><a href="/components_page.html"> Components</a></li>
```
Comment (Mohasin-Haque) [inline components/badge/badge.html]: Remove the whitespace in component

File: main.css [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  margin-left: 1.5rem;
+}
+.header__logo {
+  font-size: 2.2rem;
+}
+.logo__img {
+  width: 40px;
+}
+.header__nav {
+  display: flex;
```
Comment (Mohasin-Haque) [inline main.css]: You can also make one class for flex as you are using it again and again .. so it will reduce your work also.

**Review by Mohasin-Haque — CHANGES_REQUESTED** [review]
Your code looks good to me but you need to work on the comments. 
In the preview I saw that you haven't wrote documentation for any of the component.. Please do it. 
Thank you

File: components/avatar/avatar.html (hunk 2 above)
Comment (ark2002) [reply components/avatar/avatar.html]: hey i have made the suggested changes you can look it up

File: components/badge/badge.css [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  border-color: transparent;
+  padding: 5px 10px;
+  border-radius: 5px;
+  color: var(--white-color);
+  cursor: pointer;
+}
+
+.badge__demo-txt {
+  margin-right: 0.5rem;
+}
```
Comment (yogesh-) [inline components/badge/badge.css]: Don't use rem and px together. 
Stick to rem and define the root font size to 62.5% which makes 1rem = 10px.

File: components/avatar/avatar.css [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  font-size: 3rem;
+  padding: 1rem 2rem;
+  text-align: center;
+  vertical-align: middle;
+  background-color: var(--primary-color);
+  color: var(--white-color);
+  border-radius: 50%;
+}
+.material-icons.avatar__icon {
+  font-size: 40px;
```
Comment (xii-vi) [inline components/avatar/avatar.css]: Use relative unit.

File: components_page.html [hunk 8]
```diff
@@ -1,97 +1,97 @@
...
+      <div class="page__title">
+        <h1 class="intro__text">Components</h1>
+      </div>
+      <div class="components__container">
+        <a href="" class="component__card">
+          <div class="component__title">
+            <span>Alerts</span>
+          </div>
+          <div class="component__img-container">
+            <img src="/assests/alerts.svg" class="component__img" />
```
Comment (yogesh-) [inline components_page.html]: standard practice, always put alt_text in the image tag

File: components/avatar/avatar.css (hunk 7 above)
Comment (ark2002) [reply components/avatar/avatar.css]: hey ! its an icon used from the material icons stylesheet so thought using a absolute unit for it was the better way to work with

File: components/avatar/avatar.css [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+.busy {
+  background-color: var(--busy-color);
+}
+.notif {
+  background-color: var(--notif-color);
+  text-align: center;
+  font-family: var(--secondary-font-family);
+  font-weight: 600;
+  color: var(--white-color);
+}
```
Comment (arjunbharti) [inline components/avatar/avatar.css]: It would be better if you write proper name just like you did above.

**Review by xii-vi — COMMENTED** [review]
Use relative unit if it is greater than 10px.

**Review by yogesh- — CHANGES_REQUESTED** [review]
Minor changes suggested, otherwise looks good to me.

File: components/badge/badge.css [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+.badge--large {
+  font-size: 2.2rem;
+  padding: 2.5px 10px;
+}
+
+.badge--position {
+  margin: -10px;
+}
```
Comment (arjunbharti) [inline components/badge/badge.css]: Try to stick to one naming convention. Here you are using '--' and in the above classes you have used '__'
It's bit confusing

Comment (ark2002) [reply components/badge/badge.css]: i have used '__' for sub-elements of the block while I have used '--' for the specific attributes like sizes , positions etc.

File: components/badge/badge.css (hunk 6 above)
Comment (ark2002) [reply components/badge/badge.css]: hey i have made the suggested changes can you look it up

File: components/avatar/avatar.css (hunk 7 above)
Comment (ark2002) [reply components/avatar/avatar.css]: made the changes

**Review by arjunbharti — APPROVED** [review]
Looks fine now!

**Review by Mohasin-Haque — APPROVED** [review]

**Comment by xii-vi** [comment]
It's good now, you can merge

**Pull request merged**